### PR TITLE
Soulstones can now pull a random player if the target ghosts

### PR DESCRIPTION
**Pull request by ghost**
If the target mob ghosts, using the soulstone gives ghosts with the appropriate flag a chance to opt in with one of those "Do you want to play a Shade, please choose quickly!" messages.

The new ghost pull be put in control of the corpse, and then turned into a shade (so they'll retain the name of the original target mob).

This applies to the sac rune as well.

Flag is currently BE_CULTIST

### CODE REVIEW

**Comment by PKPenguin321** [comment]
:+1: fuck people that ghost when they're killed by a cult

**Comment by WJohn** [comment]
:+1: x :100:

**Comment by GunHog** [comment]
The FUN has been <b>INCREASED!</b>

**Comment by Robustin** [comment]
Long overdue common sense change. Let's the cult grow without getting fucked by 3-4 ghosted sacrifices and keeps inexperienced players, who didn't know that they should stay in bodies when killed by cult, from being locked out of the round. Win for cult, win for cult victims.

**Comment by Jordie0608** [comment]
A good change, but I'm not very fond of using BE_ALIEN for this.

**Comment by Fayrik** [comment]
BE_CULTIST is the flag we should be using for this, because regardless of who's doing it, shades tend to have a cult like gameplay dynamic.
But if BE_WIZARD is that important, then have it work for either.

**Comment by Dorsisdwarf** [comment]
BE_CULTIST isn't that overpopulated, so it's probbably fine to use

**Comment by Firecage** [comment]
I am guessing it will use the same popup box as the other stuff? Aka we can accidentally press yes or no while typing.

**Comment by GunHog** [comment]
Consider using the notify_ghosts() system, providing a join link in the message. No pop-up, zero risk of getting AFK players, and respecting preference flags is optional. (bans must still be respected, however - make sure banned people do not get a notification!)

File: code/game/gamemodes/wizard/soulstone.dm [hunk 1]
```diff
@@ -237,3 +240,37 @@
 	if(vic)
 		U << "<span class='info'><b>Capture successful!</b>:</span> [T.real_name]'s soul has been ripped from their body and stored within the soul stone."
 		U << "The soulstone has been imprinted with [S.real_name]'s mind, it will no longer react to other souls."
+
+
+/obj/item/device/soulstone/proc/getCultGhost(obj/item/device/soulstone/C, mob/living/carbon/human/T, mob/U)
+	var/list/candidates = get_candidates(BE_ALIEN)
```
Comment (RemieRichards) [inline code/game/gamemodes/wizard/soulstone.dm]: BE_CULTIST or bust.

**Comment by Core0verload** [comment]
Why use random instead of pulling ghosted mind back?

**Comment by RemieRichards** [comment]
@Core0verload incase the person who ghosted did it on purpose, they don't want to play, so grab another ghost who probably *does* want to play.

**Comment by Core0verload** [comment]
Then why don't use cloning-like "reenter your body" message?

**Comment by RemieRichards** [comment]
@Core0verload because that player DOES NOT WANT TO PLAY, if you gave them that message and button, THEY WOULD NOT FOLLOW IT.

this is not to get people who accidentally ghosted back, it's to get someone who's actually willing to play the gamemode.

File: code/game/gamemodes/wizard/soulstone.dm [hunk 2]
```diff
@@ -237,3 +240,42 @@
...
+					consenting_candidates += candidate
+
+	sleep(50)
+
+	if(consenting_candidates.len)
+		var/client/ghost = null
+		ghost = pick(consenting_candidates)
+		if(C.contents.len) //If they used the soulstone on someone else in the meantime
+			return 0
+		if (T.client) //If the original returned in the alloted time
```
Comment (RemieRichards) [inline code/game/gamemodes/wizard/soulstone.dm]: this if and else can be condensed down to:
```
if(!T.client)
    T.client = ghost
for(var/obj/item/W in T)
    T.unEquip(W)
    init_shade(C, T, U)
    qdel(T)
```

**Comment by KorPhaeron** [comment]
@RemieRichards 

Should be ready to merge now

**Pull request merged**